Feature/moo

JulianBarinton opened the pull request.

JulianBarinton committed:
create draft for preprocessing_moo; add custom moo attributes to compute.py and csvs

JulianBarinton committed:
working on dummy pre_processing_moo currently bus and excess data columns are deleted when running the script

Bachibouzouk committed:
Fix scenario csv files

JulianBarinton committed:
define moo functions

JulianBarinton committed:
adding moo variable for dispatchable

JulianBarinton commented:
Currently, the variable moo calculation for dispatchable.csv is not called in pre_procesing_moo.py (line 131). Somehow the scenario is not called.

JulianBarinton committed:
adding opex var to moo variable calcualtions and csvs

JulianBarinton committed:
Revert "adding opex var to moo variable calcualtions and csvs"

This reverts commit f4ccb194701697758e46482ed9a9c8fa4234e711.

JulianBarinton commented:
#TODO post processing: 
Capacity exisiting; Capacity added, Capacity Total

Necessary KPIs:
- GHG emissions
- Total upfront Investment
- Total OPEX
- Total System Cost
- Land Requirement Total 
- Land Requirement Additional
- Total Water Footprint
- Total Water Production
- Total Brine Production
- Total Electricity Production

JulianBarinton commented:
#TODO Testing of MOO. Do results make sense?

JulianBarinton committed:
add scenario Aiwa and its components

JulianBarinton committed:
add scenario arusi

Bachibouzouk commented:
> Currently, the variable moo calculation for dispatchable.csv is not called in pre_procesing_moo.py (line 131). Somehow the scenario is not called.

This is because `"dispatchable moo variable calculation"`  is not the same as `""dispatchable moo indicator calculation""`

Bachibouzouk committed:
Includes dispatchables in moo prepocessing

Bachibouzouk committed:
Merge branch 'feature/carriers-over-defined' into feature/moo

Bachibouzouk commented:
@JulianBarinton - I have merged #32 into this branch as I want to avoid as much as possible merge conflicts (fortunately there was only one in compute.py with the scenarios names). The idea is that if something is not working on your side then we can quickly come back to before the merge

Bachibouzouk committed:
Remove unused variables

Bachibouzouk committed:
Add simple calculation to the list

Bachibouzouk committed:
Apply calculation by default in post-processing

It is possible to provide calculations to the post-processing module

Bachibouzouk committed:
WIP  (DELETE THIS COMMIT LATER)

For dash app to show

Bachibouzouk committed:
Update CHANGELOG

Bachibouzouk committed:
Merge branch 'feature/carriers-over-defined' into feature/moo

Bachibouzouk committed:
Re-enable commiting scenarios for this week

Bachibouzouk committed:
Add Arusi scenario

Bachibouzouk committed:
Fix KPI and output calculations for Arusi

JulianBarinton committed:
wip, remove water trcuk as dispatchable

Bachibouzouk committed:
Add carriers to bus.csv

Important for certain calculations of outputs

Bachibouzouk committed:
Fix specific system costs calculation

JulianBarinton committed:
wip create crop facade

JulianBarinton committed:
adapt crop.csv example

JulianBarinton committed:
wip crop facade

Bachibouzouk committed:
Add 'self' to class methods and indent their content properly

Also lint using black

Bachibouzouk committed:
Prepare datapackage for crop.csv

Bachibouzouk committed:
Add SimpleCrop class

Bachibouzouk committed:
Update crop.csv related files

Bachibouzouk committed:
Add a foreign key target check when inferring

Fields which, by definition,  do not hold a foreign key are excluded from
the check

Bachibouzouk committed:
Add a foreign key target check after package is build

Bachibouzouk committed:
Add missing profiles in volatile_profile.csv

Bachibouzouk committed:
Dispatchables which do not have profile are set to None instead of '1'

Bachibouzouk committed:
Update arusi datapackage

Bachibouzouk committed:
Fix aiwa dispatchable.csv

Bachibouzouk committed:
Add comprehensive error message by casting error of CSV resources

Bachibouzouk committed:
Removed unused resource

Bachibouzouk committed:
Fix crop.csv for moo

Bachibouzouk committed:
Correct banana specs and add data source

Bachibouzouk committed:
Add a single day crop test case

There is sun as dispatchable, the production is an excess sink which
should bear the price per g or kg produced as incentive to produce.

The installed capacity of the sun should be the amount of square meters
available for crop production (maybe need to be linked via constraint)

Efficiencies of crop are still negative which does not make sense

Bachibouzouk committed:
Adapt f(heat) from reference paper

The daily maximum temperature is computed and the efficiency spread to
every hour of that day.

The function can handle number of timesteps not equal to full days

Bachibouzouk committed:
Fix calculation of temperature, heat and water stress

The test should run on several days to compare with paper biomass
production.

Also here running one day after plant sewing does not mean being able to
harvest, this should be taken into consideration

JulianBarinton committed:
improving case study Aiwa; putting real world data - WIP

JulianBarinton committed:
case study AIWA WIP

JulianBarinton committed:
errogenous code_adding carrier_cost in Aiwa elements for MOO_variable_flow calucation

Bachibouzouk committed:
Change carrier_cost to resource_cost

Bachibouzouk committed:
Make resource_cost attribute mandatory for moo

Bachibouzouk committed:
Add resource_cost to other components

JulianBarinton committed:
WIP Case Study Arusi, water filtration is modelled as MIMO; however, pontentially TYPEMAPPING is missing; The names of buses and conversion factors have been changed, which might cause problems

Bachibouzouk committed:
Merge branch 'feature/moo' of github.com:rl-institut/oemof-tabular-plugins into feature/moo

Bachibouzouk committed:
Add resource_cost to mimo for aiwa

JulianBarinton committed:
Arusi WIP

Bachibouzouk committed:
Fix storage.csv for aiwa

Bachibouzouk committed:
Fix volatile.csv for aiwa

Bachibouzouk committed:
Merge branch 'feature/moo' of github.com:rl-institut/oemof-tabular-plugins into feature/moo

Bachibouzouk committed:
Logging resource name in case of exception in pre-processing

JulianBarinton committed:
updating case study Arusi; problem with mimo water filration remains, potentially missing typemap

JulianBarinton committed:
fxing case study Arusi; service water provision has been missing

Bachibouzouk committed:
Add template for water components

Bachibouzouk committed:
Add easy import of WEFE typemap

Bachibouzouk committed:
Remove oemof-tabular foreign keys inference

This would flag the water component as chp and add fuel and heat bus as
foreign keys. OTP foreign key inference can anyways detect all foreign
keys without a-priori knowledge

Bachibouzouk committed:
Include tabular typemap from wefe package directly

Bachibouzouk committed:
Add test for water component

For the moment it is only a smoketest, running the scenario without
errors

Bachibouzouk committed:
Import TYPEMAP from wefe

Bachibouzouk committed:
Merge pull request #35 from rl-institut/feature/water-components

Feature/water components

Bachibouzouk committed:
Refine water pump Facade

JulianBarinton committed:
adding water components to case study Arusi;capacity_cost dissapears for water_filtration

Bachibouzouk committed:
Fix capacity_cost error

Bachibouzouk committed:
Prevent overwriting capacity_cost or marginal_cost if those are nan

JulianBarinton committed:
Adding Arusi case study settings for Diesel Generator

JulianBarinton committed:
WIP ARUSI COST Optimzation only

Commit:
add f_solar to crop model, update global_specs

Bachibouzouk committed:
Add a warning if the crop_type is not within the crop dict

Bachibouzouk committed:
Rename crop input and output busses

Bachibouzouk committed:
Add optional output bus for non-edible biomass production of crop

JulianBarinton committed:
Case Studies Colombia final definition of csv elements

Bachibouzouk committed:
Add raw output to the console

Bachibouzouk committed:
SimpleCrop basic working model

Bachibouzouk committed:
Update changelog

Bachibouzouk committed:
Merge pull request #33 from rl-institut/feature/simple_crop_model

Feature/simple crop model

JulianBarinton committed:
add full volatile and load profiles for case studies colombia, wrong csv seperator

JulianBarinton committed:
fix csv separator issue

JulianBarinton committed:
adapt bus column crop.csv; change time-index to timeindex in crop.py; volatile profile still cannot be read correctly

dfssf

Bachibouzouk committed:
Add a time column duplicate of timeindex

THe problem of timeindex, is that it can currently not be used as a
sequence itself.
A mapping to time_index was missing in crop.csv
Format of sowing_date and harvest_date were wrong (need MM-DD)

JulianBarinton committed:
updating unit of timeseries

Commit:
minor corrections to SimpleCrop

Commit:
minor corrections again for SimpleCrop

Commit:
WIP: Inverter Facade and test_inverter scenario

Commit:
WIP: additional stuff for inverter facade

JulianBarinton committed:
model pv as type volatile

JulianBarinton committed:
adding rainwater-harvesting as volatile resource

JulianBarinton committed:
adding condensed water as volatile; distinguishing aiwa24 and aiwa8760 for saving time while testing

JulianBarinton committed:
adding renewable ninja pv and wind turbine profile to aiwa case

JulianBarinton committed:
adding wind-turbine as volatile producer

JulianBarinton committed:
update parameters conversion.csv

JulianBarinton committed:
revise and update values dispatchable.csv

JulianBarinton committed:
review and update values in volatile.csv

JulianBarinton committed:
revise and update value storage.csv

JulianBarinton committed:
updating and revising values of mimo.csv (Scenario Aiwa24); noting that sw-ro is used (see Sankey) but not invested in

JulianBarinton committed:
revise and update values for water filtration and water pumps; water filtration is refering to microfilration; water pump costs and impacts are estimated; have to improved later; on review of bbuses, loads, excess, Aiwa24 complete

JulianBarinton committed:
adding revised compoennts to aiwa8760

JulianBarinton committed:
adding components for Arusi24 and Arusi365, maybe there was a change for Aiwa too

Bachibouzouk committed:
Add total ghg_emissions

Bachibouzouk committed:
Add dispatchables

JulianBarinton committed:
update Aiwa test cases

JulianBarinton committed:
add and update result units

JulianBarinton committed:
add description for ghg emission calculation & remove specifc system cost from kpis as it does not make senses for multi resource systems

JulianBarinton committed:
remove dispatchables table from results dash app, as it does not caclulate accumlated flow; but some kind of capacity (maxium power, production, which does not makes much sense for most dispatchables

JulianBarinton committed:
latest results aiwa_24

JulianBarinton committed:
add unit for water-filtration-system, activate MOO; first test: annuity is not calcualted correctly; no change in results when changinge moo weights
